quantuminspire/util/configuration.py

```diff
@@ -196,4 +197,19 @@ async def _fetch_team_member_id(host: str, access_token: str) -> int:
 
     @classmethod
     def get_team_member_id(cls, host: str, access_token: str) -> int:
-        return asyncio.run(cls._fetch_team_member_id(host, access_token))
+        return asyncio.run(cls._validate_token_and_retrieve_team_member_id(host, access_token))
+
+    @classmethod
+    async def _validate_token_and_retrieve_team_member_id(cls, host: str, access_token: str) -> int:
+        await cls._wait_until_token_becomes_valid(access_token)
+        return await cls._fetch_team_member_id(host, access_token)
+
+    @staticmethod
+    async def _wait_until_token_becomes_valid(access_token: str) -> None:
+        decoded_token = jwt.decode(access_token, options={"verify_signature": False})
+        token_issued_at = int(decoded_token["iat"])
+
+        while True:
+            await asyncio.sleep(1)
```

Comment:
we can also get the time difference between now and `iat` and wait for that amount of time, right?